### ISSUE TITLE
Create a Docker image

### ISSUE THREAD

**Issue by frans-ebu** [ebu/ebu-tt-live-toolkit#234]
Suggestion to create a Docker image to help users install their own copy easily.

**Comment by nigelmegitt** [comment]
Are there any licensing or pricing impacts? I'm pretty comfortable with the idea, just don't want any unpleasant unintentional consequences!

**Comment by frans-ebu** [comment]
Good question. I thought not. It is true that the Docker license itself is Apache and not BSD-3. But the Docker image is just an extra for us, it is not mandatory for using the code. So no problem?

http://54.71.194.30:4015/faq/

**Comment by kozmaz87** [comment]
Apache is compatible with us so yeah this could be a good idea.

**Comment by kozmaz87** [comment]
It has been ages since I played with Docker... My old friend here we meet again :)

**Comment by nigelmegitt** [comment]
I would expect the Docker license to be independent of our license so I'm not worried from that perspective. I'm worried more if there's a license or pricing requirement for anyone wanting to use Docker to get the image and make it run. I think from a quick scan that it is just a single one-off cloud license for Docker and then anyone can use it, but I'm not familiar with the platform so I'm not sure if there are any "gotchas" to look out for.

**Comment by frans-ebu** [comment]
I am not aware of any "gotchas". And I don't think you need to pay a cloud license (you don't have to use their cloud service!).

**Comment by nigelmegitt** [comment]
Even better! Does EBU already use Docker?

**Comment by frans-ebu** [comment]
Yes, for several projects. We will be using it for https://ebu.io/qc soon, too.

**Comment by nigelmegitt** [comment]
In that case, I see no reason not to go ahead. Do you want to "make it happen" @frans-ebu ?

**Comment by frans-ebu** [comment]
Yes, and no. I want to, but I see it as low prio (see milestone) and as I am not an expert on this (but I do have a very nice book on it ;-)), I will not make it happen any time soon.

**Comment by nigelmegitt** [comment]
OK, fine with me. Is there a particular reason why you think we should add it?

**Comment by frans-ebu** [comment]
Because it can provide a lower threshold for users to try out the code (keeps the set up well insulated from any other code; easy to deploy).

**Comment by nigelmegitt** [comment]
Right, understood - I was wondering if you had a specific date-related reason. Thanks!

**Comment by tairt** [comment]
I would fully support this. This would ease quick testing of what the components can do.

**Comment by nigelmegitt** [comment]
Although this would help with getting runnable code it is also worth checking out the configurator branch and building the documentation to take a look at it - I think the configuration scheme is now much easier to use to create sensible components and see what they can do.

If anyone wants to put some time and effort into addressing this particular issue I would be happy for that to happen, I should add. It's not a priority for me at the moment.

**Comment by spoeschel** [comment]
For a Docker image any dependencies/cache files/repo files that are not needed in such a case should not be included into the image (or removed during the same `RUN` command, to not be included into the same layer). For example the following components are not required then:

- node.js, including the `nunjucks` package; used for UI aspects only
- the Python packages listed in `requirements.txt` (except `-e .` for the project itself); used for documentation/testing only
- the `docs` and `testing` repo folders
- the `pip` cache
- the `pip` and `wheel` modules, after everything has been installed
